common/djangoapps/contentserver/middleware.py

```diff
@@ -55,16 +56,32 @@ def process_request(self, request):
             except (ItemNotFoundError, NotFoundError):
                 return HttpResponseNotFound()
 
+            # Set the basics for this request.
+            newrelic.agent.add_custom_parameter('contentserver.course_key', loc.course_key)
+            newrelic.agent.add_custom_parameter('contentserver.path', loc.path)
+
+            # Figure out if this is a CDN using us as the origin.
+            is_from_cdn = StaticContentServer.is_cdn_request(request)
+            newrelic.agent.add_custom_parameter('contentserver.from_cdn', 1 if is_from_cdn else 0)
+
+            # Check if this content is locked or not.
+            locked = self.is_content_locked(content)
+            newrelic.agent.add_custom_parameter('contentserver.locked', 1 if locked else 0)
+
             # Check that user has access to the content.
             if not self.is_user_authorized(request, content, loc):
+                newrelic.agent.add_custom_parameter('contentserver.authorized', 0)
                 return HttpResponseForbidden('Unauthorized')
 
+            newrelic.agent.add_custom_parameter('contentserver.authorized', 1)
+
             # Figure out if the client sent us a conditional request, and let them know
             # if this asset has changed since then.
             last_modified_at_str = content.last_modified_at.strftime(HTTP_DATE_FORMAT)
             if 'HTTP_IF_MODIFIED_SINCE' in request.META:
                 if_modified_since = request.META['HTTP_IF_MODIFIED_SINCE']
                 if if_modified_since == last_modified_at_str:
+                    newrelic.agent.add_custom_parameter('contentserver.not_modified', 1)
                     return HttpResponseNotModified()
```

Comment:
With this and the authorized header -- is that information we can already get by looking at the response code? If not, is it something we can collapse into a single metric, so we could facet on the kind of response they got?